File: app/src/main/java/zerochanimage/FindLink.java

package zerochanimage;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.edge.EdgeOptions;


import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.TimeUnit;

public class FindLink {
    EdgeOptions options;
    WebDriver driver;

    FindLink() {
        WebDriverManager.edgedriver().setup();
        options = new EdgeOptions();
        options.addArguments("--remote-allow-origins=*");
        //options.setHeadless(true);

        driver = new EdgeDriver(options);

        driver.manage().window().minimize();

    }

    public void Login() {

        driver.get("https://www.zerochan.net/login");
        WebElement login = driver.findElement(By.name("name"));
        WebElement password = driver.findElement(By.name("password"));
        WebElement button = driver.findElement(By.name("login"));;


        login.sendKeys("hieu555x");
        password.sendKeys("Trung_hieu9@");
        button.click();

        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
        //#endregion
    }

    public ArrayList<String> getAllImageLink(String Url) {
        ArrayList<String> listLink = new ArrayList<>();

        driver.get(Url);

        Document document = Jsoup.parse(driver.getPageSource());

        Element ulElement = document.getElementById("thumbs2");

        List<Element> listElement = ulElement.getElementsByTag("li");

        for (Element e : listElement) {

            Elements aElement = e.getElementsByTag("a");
            for (Element element : aElement) {
                if (element.absUrl("href").contains("static")) {
                    listLink.add(element.absUrl("href"));

                    //System.out.println(element.absUrl("href"));
                }
            }

        }
        return listLink;
    }

    public int getPageIndex(String Url) {
        driver.get(Url);
        WebElement pageIndex = driver.findElement(By.className("pagination"));
        String s = pageIndex.getText();
        int Index = 0;
        for (int i = 0; i < s.split(" ").length; i++) {
            try {
                Index = Integer.valueOf(s.split(" ")[i]);
            } catch (Exception ex) {
                Index = Index;
            }
        }
        return Index;
    }

    public void quitChrome() {
        driver.quit();
    }
}

File: DataFile/findLink.cs
//using OpenQA.Selenium;
//using OpenQA.Selenium.Chrome;
//using OpenQA.Selenium.Edge;
//using System;
//using System.Collections;
//using System.Collections.Generic;
//using System.Linq;
//using System.Reflection.Metadata;
//using System.Text;
//using System.Threading.Tasks;

//namespace ZerochanImageSelenium.DataFile
//{
//    internal class findLink
//    {
//        EdgeOptions options;
//        WebDriver driver;

//        public findLink()
//        {
//            options = new EdgeOptions();
//            options.AddArgument("--remote-allow-origins=*");

//            driver = new EdgeDriver(options);

//            //driver.Manage().Window.Minimize();
//        }

//        public void Login()
//        {
//            driver.Navigate().GoToUrl("https://www.zerochan.net/login");
//            WebElement login = (WebElement)driver.FindElement(By.Name("name"));
//            WebElement password = (WebElement)driver.FindElement(By.Name("password"));
//            WebElement button = (WebElement)driver.FindElement(By.Name("login"));

//            login.SendKeys("hieu555x");
//            password.SendKeys("Trung_hieu9@");
//            button.Click();

//            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(5);
//        }

//        public ArrayList getAllImageLink(string url)
//        {
//            ArrayList listLink = new ArrayList();

//            driver.Navigate().GoToUrl(url);



//            return listLink;
//        }
//    }
//}
